lib/matplotlib/figure.py

```diff
@@ -682,11 +682,10 @@ def set_size_inches(self, *args, **kwargs):
         matplotlib.Figure.get_size_inches
         """
 
-        forward = kwargs.get('forward', False)
-        if len(args) == 1:
-            w, h = args[0]
-        else:
-            w, h = args
+        # the width and height have been passed in as a tuple to the first
```

Comment:
True, and it has its good points; but I'm not sure it's worth the added API complexity to support this form everywhere.  set_xlim and set_ylim are higher level than set_size_inches, so it is less disruptive to simplify the latter than the former.
As for set_size_inches versus set_size: I suspect that like much in mpl, it is an accident of the way it evolved.  Whoever put it in decided to make it very clear what the units are.